crates/next-api/src/app.rs

```diff
@@ -865,7 +866,33 @@ impl AppEndpoint {
                 }
                 let client_shared_availability_info = client_shared_chunk_group.availability_info;
 
-                let client_references = client_reference_graph(Vc::cell(vec![rsc_entry_asset]));
+                let client_references = {
+                    let ServerEntries {
+                        server_component_entries,
+                        server_utils,
+                    } = &*find_server_entries(rsc_entry).await?;
+
+                    let mut client_references = client_reference_graph(
+                        server_utils.clone(),
+                        VisitedClientReferenceGraphNodes::empty(),
+                    )
+                    .await?
+                    .clone_value();
+
+                    for module in server_component_entries
+                        .iter()
+                        .map(|m| Vc::upcast::<Box<dyn Module>>(*m))
+                        .chain(std::iter::once(rsc_entry))
+                    {
+                        let current_client_references =
+                            client_reference_graph(vec![module], client_references.visited_nodes)
+                                .await?;
+
+                        client_references.extend(&current_client_references);
+                    }
+                    client_references
+                };
+                let client_references_cell = client_references.clone().cell();
```

Comment:
```suggestion
                let client_references_cell = client_references.cell();
```

Don't we already own the value? Do we need to clone it?

Reply:
Various calls further down need a `Vc<>` to pass as an argument but the fields in `client_references` are also used multiple times.
So this clone isn't needed because the value is not owned, but because `cell()` consumes it and we still want to read `client_references`.